include/stdexec/execution.hpp

```diff
@@ -5736,13 +5736,428 @@ namespace stdexec {
+              throw;
+            }
+            __object_pointer_ = __pointer;
+        }
+
+        __t(const __t& __other) {
+          (*__other.__vtable_)(__copy_construct, this, __other);
+        }
+
+        __t& operator=(const __t& __other) {
+          (*__other.__vtable_)(__copy_construct, this, __other);
+          return *this;
+        }
+
+        __t(__t&& other) noexcept
+        : __vtable_{std::exchange(other.__vtable_, nullptr)}
+        , __object_pointer_{std::exchange(other.__vtable_, nullptr)} {}
+
+        __t& operator=(__t&& other) noexcept {
+          __vtable_= std::exchange(other.__vtable_, nullptr);
+          __object_pointer_ = std::exchange(other.__vtable_, nullptr);
+          return *this;
+        }
+
+        ~__t() {
+          __reset();
+        }
+
+        void __reset() noexcept {
+          if (__vtable_) {
+            (*__vtable_)(__delete, this);
+            __object_pointer_ = nullptr;
+            __vtable_ = nullptr;
+          }
+        }
+
+       private:
+        friend const _Vtable* tag_invoke(__get_vtable_t, const __t& __self) noexcept {
+          return __self.__vtable_;
+        }
+
+        friend void* tag_invoke(__get_object_pointer_t, const __t& __self) noexcept {
+          return __self.__object_pointer_;
+        }
+
+        template <class _T>
+          friend void tag_invoke(__delete_t, __type<_T>, __t& __self) noexcept {
+            using _Alloc = typename 
+                std::allocator_traits<_Allocator>::template rebind_alloc<_T>;
+            if (__self.__object_pointer_) {
+              _Alloc __alloc{__self.__allocator_};
+              _T* __pointer = reinterpret_cast<_T*>(std::exchange(__self.__object_pointer_, nullptr));
+              std::allocator_traits<_Alloc>::destroy(__alloc, __pointer);
+              std::allocator_traits<_Alloc>::deallocate(__alloc, __pointer, 1);
+            }
+          }
+
+        template <class _T>
+          friend void tag_invoke(__copy_construct_t, __type<_T>, __t& __self, const __t& __other) {
+            using _Alloc = typename 
+                std::allocator_traits<_Allocator>::template rebind_alloc<_T>;
+            _Alloc __alloc{__self.__allocator_};
+            _T* __pointer = std::allocator_traits<_Alloc>::allocate(__alloc, 1);
+            try {
+              std::allocator_traits<_Alloc>::construct(
+                  __alloc, __pointer,  *reinterpret_cast<const _T*>(__other.__object_pointer_));
+              __self.__reset();
+              __self.__object_pointer_ = __pointer;
+              __self.__vtable_ = __other.__vtable_;
+            } catch (...) {
+              std::allocator_traits<_Alloc>::deallocate(__alloc, __pointer, 1);
+              throw;
+            }
+          }
+
+        const __vtable_t* __vtable_{nullptr};
+        void* __object_pointer_{nullptr};
+        [[no_unique_address]] _Allocator __allocator_{};
+      };
```

Comment:
It should be possible to share implementation between the different storage types, particularly the movable and copyable ones.

Reply:
I will try!